src/context/VideoFilterContext.jsx

```diff
@@ -0,0 +1,38 @@
+import { useReducer, createContext, useContext } from "react";
+
+const VideoFilterContext = createContext();
+
+const VideoFilterProvider = ({ children }) => {
+  const initialFilterState = {
+    showCategories: [],
+  };
+  const filter_reducer_fn = (state, { type, payload }) => {
```

Comment:
Suggestion: As you are using camelCasing everywhere why have you used snakeCase here?

Reply:
yes realised it later, changed